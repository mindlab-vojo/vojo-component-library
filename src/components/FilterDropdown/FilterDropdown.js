import React, { useState } from 'react'
import PropTypes from 'prop-types'

import Typography from '../Typography'
import IconManager from '../IconManager'
import * as FilterDropdown from './FilterDropdown.style'

import ThemeConsumer from '../../style/ThemeManager/ThemeProvider'

const FilterDropdownComponent = ({
  defaultOption,
  options,
  onSelectOption,
  placeholder,
}) => {
  const [isSelected, setIsSelected] = useState(Boolean(defaultOption))

  const renderComponent = theme => {
    
    const currentColor = isSelected 
      ? theme.colors.blackColor
      : theme.colors.whiteColor
    return (
      <FilterDropdown.Wrapper
        isSelected={isSelected}
        theme={theme}>
        <FilterDropdown.Text>
          <Typography 
            color={currentColor}
            tag="p"
            type="default">
            TESTE
          </Typography>
        </FilterDropdown.Text>
        <FilterDropdown.Icon>
          <IconManager 
            icon="ArrowDown"
            fill={currentColor}
          />
        </FilterDropdown.Icon>
      </FilterDropdown.Wrapper>
    )}

  return (
    <ThemeConsumer>
      {({ theme }) => renderComponent(theme)}
    </ThemeConsumer>
  )
}

FilterDropdownComponent.propTypes = {
  defaultOption: PropTypes.any,
  options: PropTypes.object,
  onSelectOption: PropTypes.func,
  placeholder: PropTypes.string,
}

export default FilterDropdownComponent
