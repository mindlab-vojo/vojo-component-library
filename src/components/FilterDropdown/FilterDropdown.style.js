import styled from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  position: relative;
  width: fit-content;
`

export const Input = styled.div`
${props => props.isSelected 
    ? `
    background-color: ${props.theme.colors.blackColor};
    ` : `
    background-color: ${props.theme.colors.whiteColor};
    `
}
  align-items: center;
  box-sizing:border-box;
  cursor: pointer;
  border: 2px solid ${props => props.theme.colors.blackColor};
  border-radius: 10px;
  display: flex;
  padding: 5px 12px;
  width: 100%;
`

export const Text = styled.div``

export const Icon = styled.div`
  margin-left: 5px;
  margin-top: 7px;
`
export const Dropdown = styled.div`
  background-color: ${props => props.theme.colors.blackColor};
  border-bottom-left-radius: 10px;
  border-bottom-right-radius: 10px;
  margin-top: -10px;
  overflow: hidden;
  position: absolute;
  top: 54px;
  width: 100%;
`
export const Item = styled.div`
  background-color: ${props => props.theme.colors.blackColor};
  cursor: pointer;
  padding: 8px 14px;
  width: 100%;

  &:hover {
    background-color: ${props => props.theme.colors.darkGreyColor};
  }
`