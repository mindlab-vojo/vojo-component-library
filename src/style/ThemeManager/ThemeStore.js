import { vojo } from './../themes'

class ThemeStore {

  constructor(theme) {
    this.defaultTheme = vojo
    this.theme = theme ? theme : this.defaultTheme
  }

  configure = ({ theme }) =>  {
    if (theme) this.configureTheme(theme)
    else this.configureTheme(this.defaultTheme)
  }

  configureTheme = (theme) => {
    this.theme = theme
  }

  modify = (modifiedProps) => {
    const currentTheme = this.theme
    return { currentTheme, modifiedProps }
  }

  getTheme = () => this.theme

}

export default ThemeStore