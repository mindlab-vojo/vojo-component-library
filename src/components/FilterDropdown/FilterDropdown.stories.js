import React from 'react';
import { withKnobs, text } from '@storybook/addon-knobs';
import { withA11y } from '@storybook/addon-a11y';
import { action } from '@storybook/addon-actions'

import FilterDropdown from '.'; 
import { ThemeProvider } from "../../../src";
import categories from '../../../.storybook/categories';

export const component = () => {

  return (
    <FilterDropdown/>
  )
}

export default {
  title: categories.Dropdowns,
  component: component,
  decorators: [
    withKnobs, 
    withA11y, 
    (story) => <ThemeProvider>{ story() }</ThemeProvider>,
    (story) => <div style={{ width: "150px" }}>{ story() }</div>,
  ],
}

component.story = {
  name: 'FilterDropdown',
};
  