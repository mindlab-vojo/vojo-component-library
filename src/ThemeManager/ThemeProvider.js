import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  render() {
    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer