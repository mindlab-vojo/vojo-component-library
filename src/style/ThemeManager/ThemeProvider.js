import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  getThemeFont = () => {
    const { themeConfig } = this.state
    return themeConfig.font
  }

  render() {

    const GlobalFont = this.getThemeFont()

    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        <GlobalFont/>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer