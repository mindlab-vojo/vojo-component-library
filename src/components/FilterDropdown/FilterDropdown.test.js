import React from 'react';
import { render } from '@testing-library/react';

import FilterDropdown from './';
import { ThemeProvider } from './../../style/ThemeManager/ThemeProvider';

test('Test FilterDropdown component render', () => {
  const fakeFn = jest.fn()
  const component = render(
    <ThemeProvider>
      <FilterDropdown 
        onSelectOption={fakeFn}
        options={[{ id: "1", name: "Test", value: "test-1" }]}
        placeholder={"Test"}
      />
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
