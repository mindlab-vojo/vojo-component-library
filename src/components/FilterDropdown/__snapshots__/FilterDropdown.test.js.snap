// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FilterDropdown component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxjAm ilhmbz"
      >
        <div
          class="sc-AxirZ rfaeh FilterDropdown"
        >
          <div
            class="sc-AxiKw bOTXee FilterDropdown__Text"
          >
            <p
              class="sc-AxheI feOFeg"
            >
              Test
            </p>
          </div>
          <div
            class="sc-AxhCb gkOWJK FilterDropdown__Icon"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                fill="#212529"
              />
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm ilhmbz"
    >
      <div
        class="sc-AxirZ rfaeh FilterDropdown"
      >
        <div
          class="sc-AxiKw bOTXee FilterDropdown__Text"
        >
          <p
            class="sc-AxheI feOFeg"
          >
            Test
          </p>
        </div>
        <div
          class="sc-AxhCb gkOWJK FilterDropdown__Icon"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
              fill="#212529"
            />
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
